addons/Hierarchical_Table_Of_Contents/src/presenter.js

```diff
@@ -262,7 +278,11 @@ function AddonHierarchical_Table_Of_Contents_create() {
         presenter.treeID = presenter.configuration.ID + (isPreview ? "Preview" : "");
         presenter.$view.find("div").first().attr('id', presenter.treeID);
 
-        presenter.setVisibility(presenter.configuration.isVisible || isPreview);
+        if (presenter.hasOwnProperty('isVisible')) {
+            presenter.setVisibility(presenter.isVisible);
```

Comment:
Po pierwsze, skoro to jest metoda initialize, to wątpię, żeby ten warunek był kiedykolwiek spełniony. Sprawdź to proszę i usuń jeśli jest zbędne. Po drugie, należałoby zdefiniować property isVisible na początku pliku razem z innymi zmiennymi, zamiast go tworzyć wewnątrz setVisibility. Przy okazji rozwiązuje to też problem konieczności sprawdzenia, czy presenter ma to property (gdyby się okazało, że ten warunek jest jednak potrzebny, możesz dać domyślnie wartość null dla isVisible, żeby można było rozróżnić czy już było ustawione czy nie).